File: app/src/main/java/com/bryan/platform/model/response/Result.java
package com.bryan.platform.model.response;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.google.gson.annotations.SerializedName;

/**
 * 统一响应封装数据模型（Java 版）
 * 对应后端 Result 泛型类
 */
public class Result<T> {

    @SerializedName("code")
    private int code;

    @SerializedName("message")
    private String message;

    @SerializedName("data")
    @Nullable
    private T data;

    public Result() {
    }

    public Result(int code, String message, @Nullable T data) {
        this.code = code;
        this.message = message;
        this.data = data;
    }

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    @Nullable
    public T getData() {
        return data;
    }

    public void setData(@Nullable T data) {
        this.data = data;
    }

    public boolean isSuccess() {
        return this.code == 200;
    }

    @NonNull
    @Override
    public String toString() {
        return "Result{" +
                "code=" + code +
                ", message='" + message + '\'' +
                ", data=" + data +
                '}';
    }
}

File: app/src/main/java/com/bryan/platform_android/model/response/Result.kt
package com.bryan.platform_android.model.response // 使用您的命名空间

import com.google.gson.annotations.SerializedName

/**
 * 统一响应封装数据模型
 * 对应后端 Result 泛型类
 */
data class Result<T>(
    @SerializedName("code")
    val code: Int,
    @SerializedName("message")
    val message: String,
    @SerializedName("data")
    val data: T? // T 可以是 String (登录成功返回 token), User (注册成功返回用户), 或 Page<User> 等
) {
    // 判断请求是否成功
    fun isSuccess(): Boolean {
        // 根据后端 ErrorCode.SUCCESS 的 code 值来判断
        return this.code == 200 // 假设成功码为 200
    }
}